### ISSUE TITLE
Extract variable and method on arguments of decorator which precedes function definition doesn't create required definitions.

### ISSUE THREAD

**Issue by IntoLighter** [microsoft/pylance-release#3153]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

Expected: Select value that I would like to extract, press extract. Values that were selected move to a definition statement, and I am given opportunity to rename the extracted component.

Actual: Value of selection substitutes by default extraction component name. No statements are created.

## Steps to reproduce:

Create a function with decorator and pass it some argument, then extract this argument. Function for example:

@staticmethod(5)
def a():
    pass

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.5
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
Output doesn't contain anything related to issue. And there is no way to paste anything to issue reporter.
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

testing
• pytestArgs: "<placeholder>"
• pytestEnabled: true

```

</p>
</details>

Extension version: 2022.12.0
VS Code version: Code 1.70.0 (Universal) (da76f93349a72022ca4670c1b84860304616aaa2, 2022-08-04T04:38:55.829Z)
OS version: Darwin arm64 21.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 2|
|Memory (System)|16.00GB (0.06GB free)|
|Process Argv|--crash-reporter-id 6c731971-c05a-4f5a-9769-babf9ff05d06|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by bschnurr** [comment]
I'm see this... can you provide an example of what you expect? 
```python
def new_func():
    return 5

@staticmethod(new_func())
def a():
 pass
```

do you expect to see a new variable introduced?
```python
i = 4

@staticmethod(i)
def a():
 pass
```

**Comment by IntoLighter** [comment]
I'm expecting to see what you have written for both cases. If you didn't understand, I get now:
```py
@staticmethod(new_func())
def a():
 pass
```

```py
@staticmethod(new_var)
def a():
 pass
```

**Comment by bschnurr** [comment]
so you get `new_func()` but not the definition `def new_func():  pass` ?

**Comment by IntoLighter** [comment]
@bschnurr, yes. This is the problem of the issue.

**Comment by bschnurr** [comment]
what platform?

i can't seem to repo the issue. i thought there might be a problem when its the first line of code in a file but it seems to be working for me.

**Comment by bschnurr** [comment]
I see it now. 'extract variable'

**Comment by debonte** [comment]
This issue was fixed in Pylance 2022.8.31, but we neglected to close it.
